### PR TITLE
refactor: remove multiple machinery global lists, use SSmachinery's machinery by type, use `all_areas` instead of `world` for all area iterations

### PR DESCRIPTION
**Pull request by Gaxeer**
## What Does This PR Do
- Replace `in world` to `in GLOB.all_areas` for area iterations
- Add 2 new lists to SSmachines: `all_machinery` and `machiney_by_type`. `machinery_by_type allows to more efficiently get subset of machinery.
- Replace `GLOB.machinery` with `get_machinery_of_type` calls, which will speed up iteration that were using whole machinery list, instead of subset.
- Remove:
    - GLOB.airlocks
    - GLOB.prisoncomputer_list
    - GLOB.navigation_computers
    - GLOB.apcs
    - GLOB.air_alarms
    - GLOB.power_monitors
    - GLOB.all_vent_pumps
    - GLOB.deliverybeacons

 And replace them with `get_machinery_of_type(specific_type)`

## Why It's Good For The Game
Good for coders, that won't need to create millions of global lists for specific machinery subtype.
Players won't notice, probably.

## Testing
Compiles, nothing special. I am too lazy to fix it locally, so we run Testmerge on downstream to catch runtimes.
Performance shouldn't change much, except cases where GLOB.machinery were used instead of subset. 
Maybe will benchmark later

## Changelog
Nothing player facing

### CODE REVIEW

**Review by AffectedArc07 — APPROVED** [review]
Approved what I own

**Review by lewcc — COMMENTED** [review]
Looks like this is creating some runtimes in CI, might be worth looking at (or re-running CI to see if it was ephemeral).

Also, if this goes in, it might be something worth adding to our contributor documentation since this looks like a good change to keep in mind down the line.

**Comment by DGamerL** [comment]
> Looks like this is creating some runtimes in CI, might be worth looking at (or re-running CI to see if it was ephemeral).
> 
> Also, if this goes in, it might be something worth adding to our contributor documentation since this looks like a good change to keep in mind down the line.

It's failing CI because the science burn chamber is kinda fucked on Metastation, and this PR somehow kinda fixes the logic on that or something? Hopefully after #24156 is merged it'll pass CI

File: code/game/machinery/magnetic_module.dm [hunk 1]
```diff
@@ -214,7 +214,6 @@
 /obj/machinery/magnetic_controller/LateInitialize()
 	..()
 	if(autolink)
-		// GLOB.machines is populated in /machinery/Initialize
```
Comment (Contrabang) [inline code/game/machinery/magnetic_module.dm:217]: This should still be mentioned, but have it say something like
```
The global machinery associated list is populated in /machinery/Initialize
```

File: code/modules/mob/living/silicon/robot/drone/maint_drone.dm [hunk 2]
```diff
@@ -411,10 +411,10 @@
 		return FALSE // Pretty damn hard to path through space
 
 	var/turf/target
-	for(var/obj/machinery/drone_fabricator/DF in GLOB.machines)
-		if(DF.z != z)
+	for(var/obj/machinery/drone_fabricator/drone_fabricator as anything in SSmachines.get_machinery_of_type(/obj/machinery/drone_fabricator))
+		if(atoms_share_level(drone_fabricator, src))
```
Comment (Contrabang) [inline code/modules/mob/living/silicon/robot/drone/maint_drone.dm:415]:
```suggestion
		if(!atoms_share_level(drone_fabricator, src))
```

**Review by Contrabang — CHANGES_REQUESTED** [review]
Minor things I spotted, looks pretty good

**Comment by Gaxeer** [comment]
> Looks like this is creating some runtimes in CI, might be worth looking at (or re-running CI to see if it was ephemeral).
> 
> Also, if this goes in, it might be something worth adding to our contributor documentation since this looks like a good change to keep in mind down the line.

Will add it tomorrow and finish this stuff

**Comment by Contrabang** [comment]
Any updates on this?

**Comment by Contrabang** [comment]
🫡

**Pull request closed without merging**